src/indexer.hh

```diff
@@ -163,14 +165,15 @@ struct FuncDef : NameMixin<FuncDef<V>> {
   int16_t short_name_size = 0;
   SymbolKind kind = SymbolKind::Unknown;
   SymbolKind parent_kind = SymbolKind::Unknown;
+  SymbolIdx parent = {0, Kind::Invalid};
```

Comment:
Will this increase memory usage?

Reply:
Yeah, certainly, I haven't done measures. But I am working on another idea which doesn't introduce new things to the index and seems simpler and cleaner.